Library/Homebrew/formulary.rb

```diff
@@ -446,7 +453,9 @@ def initialize(alias_path)
   class FromPathLoader < FormulaLoader
     def initialize(path)
       path = Pathname.new(path).expand_path
-      super path.basename(".rb").to_s, path
+      name = path.basename(".rb").to_s
+      name = name.split("--", 2).last if path.dirname == HOMEBREW_CACHE/"downloads"
```

Comment:
Might be nice to add a comment to explain this. I know why but not sure others will.